web/src/settings_org.js

```diff
@@ -702,29 +707,172 @@ export function discard_property_element_changes(elem, for_realm_default_setting
+
+    function get_allowed_duration_message() {
+        if (maximum_allowed_days === null) {
+            return $t(
+                {
+                    defaultMessage:
+                        "Enter a value greater than or equal to {minimum_allowed_days} days",
+                },
+                {minimum_allowed_days},
+            );
+        }
+        return $t(
+            {
+                defaultMessage:
+                    "Enter a value between {minimum_allowed_days} and {maximum_allowed_days} days",
+            },
+            {minimum_allowed_days, maximum_allowed_days},
+        );
+    }
+
+    function is_valid_time_period(time_period) {
+        if (time_period === "custom") {
+            return true;
+        }
+        if (time_period === "null") {
+            if (maximum_allowed_days === null) {
+                return true;
+            }
+            return false;
+        }
+        if (maximum_allowed_days === null && time_period >= minimum_allowed_days * 24 * 60) {
+            return true;
+        }
+        if (
+            time_period >= minimum_allowed_days * 24 * 60 &&
+            time_period <= maximum_allowed_days * 24 * 60
+        ) {
+            return true;
+        }
+        return false;
+    }
+
+    for (const key in settings_config.realm_deactivation_in_values) {
+        if (is_valid_time_period(settings_config.realm_deactivation_in_values[key].value)) {
+            delete_in_options[key] = settings_config.realm_deactivation_in_values[key];
+        }
+    }
+
+    const html_body = render_settings_deactivate_realm_modal({
+        is_admin: page_params.is_admin,
+        is_owner: page_params.is_owner,
+        development_environment: page_params.development_environment,
+        delete_in_options,
+        allowed_duration_message: get_allowed_duration_message(),
+        time_choices: time_unit_choices,
+    });
+
+    function deactivate_realm_modal_post_render() {
+        invite.set_custom_time_inputs_visibility(
+            custom_expiration_time_unit,
+            custom_expiration_time_input,
+        );
+
+        const expiration_time_in_minutes = invite.get_expiration_time_in_minutes(
+            custom_expiration_time_unit,
+            custom_expiration_time_input,
+        );
+        const valid_to_text = valid_to(expiration_time_in_minutes);
+        invite.set_expires_on_text(valid_to_text);
+
+        $("#expires_in").on("change", () => {
+            invite.set_custom_time_inputs_visibility(
+                custom_expiration_time_unit,
+                custom_expiration_time_input,
+            );
+
+            const expiration_time_in_minutes = invite.get_expiration_time_in_minutes(
+                custom_expiration_time_unit,
+                custom_expiration_time_input,
+            );
+            const valid_to_text = valid_to(expiration_time_in_minutes);
+            invite.set_expires_on_text(valid_to_text);
+        });
```

Comment:
All of the `invite.` code in here needs to be moved out to someplace more generic -- this code has nothing to do with invitations.  It's a UI widget.

Reply:
Hmm, was thinking to move the three functions `set_custom_time_inputs_visibility`, `get_expiration_time_in_minutes`,`set_expires_on_text` which are currently in `invite.js` to `settings_ui.js`. Is there any better place where these functions can be moved?

Reply:
The frontend code is not my area of expertise, but that seems like a fine place for it to me.